naucse/routes.py

```diff
@@ -211,3 +212,39 @@ def lesson(lesson, page):
     page = lesson.pages[page]
     g.vars = dict(page.vars)
     return render_page(page=page, page_wip=True)
+
+
+####### SESSION
+
+
+def session_template_or_404(run, session, page):
+    env = app.jinja_env.overlay(loader=session_template_loader)
+    run_link = str(run).split(" - ")[0]
+    name = '{}/sessions/{}/{}.md'.format(run_link, session, page)
+    try:
+        return env.get_template(name)
+    except TemplateNotFound:
+        abort(404)
+
+
+def render_session(run, session, page):
+    template = session_template_or_404(run, session, page)
+    content = Markup(template.render())
+
+    md_content = Markup(convert_markdown(content))
+
+    return render_template('session.html', content=md_content, page=page)
```

Comment:
A separate `render_session` function is not necessary. Its functionality is not that well separated from what should be in `session_page`.

This is unlike `session_template_or_404`, which has a clear purpose.

It's also unlike `render_page`, which exists as a separate function because it's called for both the "just a lesson" URL and for lessons in a run – the common code is in a shared function rather than repeated. But `render_session` is only called from one place.